test/parallel/test-fs-write-optional-params.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const common = require('../common');
+
+// This test ensures that fs.write accepts "named parameters" object
+// and doesn't interpret objects as strings
+
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+const tmpdir = require('../common/tmpdir');
+const util = require('util');
+
+tmpdir.refresh();
+
+const destInvalid = path.resolve(tmpdir.path, 'rwopt_invalid');
+const buffer = Buffer.from('zyx');
+
+function testInvalidCb(fd, expectedCode, buffer, options, callback) {
+  assert.throws(
+    () => fs.write(fd, buffer, options, common.mustNotCall()),
+    { code: expectedCode }
+  );
+  callback(0);
+}
+
+// ([ args ], index, _) signature is for .forEach() or .map()
+function testValidCb([ buffer, options ], index, _, callback) {
+  const dest = path.resolve(tmpdir.path, `rwopt_valid_${index}`);
+  fs.open(dest, 'w+', common.mustSucceed((fd) => {
+    fs.write(fd, buffer, options, common.mustSucceed((bytesWritten, bufferWritten) => {
+      const writeBufCopy = Uint8Array.prototype.slice.call(bufferWritten);
+
+      fs.read(fd, buffer, options, common.mustSucceed((bytesRead, bufferRead) => {
+        const readBufCopy = Uint8Array.prototype.slice.call(bufferRead);
+
+        assert.ok(bytesWritten >= bytesRead);
+        if (options.length !== undefined && options.length !== null) {
+          assert.strictEqual(bytesWritten, options.length);
+        }
+        if (options.offset === undefined || options.offset === 0) {
+          assert.deepStrictEqual(writeBufCopy, readBufCopy);
+        }
+        assert.deepStrictEqual(bufferWritten, bufferRead);
+        fs.close(fd, common.mustSucceed(callback));
+      }));
+    }));
+  }));
+}
+
+const testInvalid = util.promisify(testInvalidCb);
+const testValid = util.promisify(testValidCb);
+
+fs.open(destInvalid, 'w+', common.mustSucceed(async (fd) => {
```

Comment:
IMHO it's safer to ensure we don't let any never settling promises skip some tests.
```suggestion
async function runTests(fd) {
```